docs/learning/ui/AugmentingUI.md

```diff
@@ -1,10 +1,38 @@
+There are two basic ways to augment the UI of a host iTwinApp.
+
+The simplest way is to use a `UiItemsProvider` to provide definitions for Tool buttons, Status Bar items, and Widgets to add to an existing frontstage. In this scenario, as frontstage components are constructed at runtime, calls are made to all registered UiItemsProviders to gather item definitions to insert into the host applications UI. The item definitions are sorted and arranged by their itemPriority value.
+
+ The second way is for a package to provide an entire stage definition. It is recommended that this be done using the [StandardFrontstageProvider]($appui-react), which will provide an empty stage that can then be populated via UiItemsProviders.
+
+A package must have an initialize() method that registers its UiItemsProvider, as in this example:
+
+```ts
+  private static registerUiComponents(): void {
+    SampleTool.register(MyPackage.localizationNamespace);
+    GenericTool.register(MyPackage.localizationNamespace);
+
+    // register to add items to "General" usage stages"
+    UiItemsManager.register(new MyUiItemsProvider());
+  }
+
+  public static async initialize(): Promise<void> {
+    if (this._initialized)
+      return;
+
+    /** Register the localized strings for this package
+     * We'll pass the localization member to the rest of the classes in the package to allow them to translate strings in the UI they implement.
+     */
+    await IModelApp.localization.registerNamespace(MyPackage.localizationNamespace);
+    this.registerUiComponents();
+    this._initialized = true;
+  }
+```
 
 ## Adding ToolButtons, Status Bar items, and Widgets to existing application frontstage
 
-A [UiItemsProvider]($appui-abstract) is used to provide items to insert into the UI of an existing stage. When constructing the stage the UiFramework code will request item definitions from the UiItemsProvider. These calls will always include the current frontstage's Id and usage. An extension can use the info to decide which items to add. The stageId name's used by an application may not be useful unless the extension is just used in a single host app where the stage names are known. The stageUsage value is also provided, this string is typically set to one of the standard [StageUsage]($appui-abstract) enum values.
+A [UiItemsProvider]($appui-abstract) is used to provide items to insert into the UI of an existing stage. When constructing the stage the UiFramework code will request item definitions from the UiItemsProvider. These calls will always include the current frontstage's Id and usage. An package can use the info to decide which items to add. The stageId name's used by an application may not be useful unless the package is just used in a single host app where the stage names are known. The stageUsage value is also provided, this string is typically set to one of the standard [StageUsage]($appui-abstract) enum values.
```

Comment:
nit: wording seems off a bit `When constructing the stage the UiFramework code will request...` what if it said something like `When constructing the stage AppUI will request...` (I guess you could also use UiFramework if you wanted it to match the actual class)

nit: /An package/A package/

Reply:
> Overall looks good, a few minor wording or spelling comments.
> 
> One thing I was thinking while reading the docs is if we want to center the phrasing all around `packages`. While technically true that most people will want to pull these things out into packages for the most re-usability, it does seem to indicate you wouldn't use the UiItemsProviders within the application code itself. Is that what we want?

We do say that the recommended way to build a Frontstage is via UiItemsProvider. Perhaps in the doc reorg, I can emphasize providers more.